plugins/fireeye_hx/icon_fireeye_hx/actions/get_alerts_by_host_id/action.py

```diff
@@ -16,32 +14,10 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        host_id = params.get(Input.HOST_ID)
-        url = f"{self.connection.url}/hx/api/v3/alerts/filter"
-
-        payload = json.dumps({"agent._id": [host_id]})
-
-        response = self.connection.session.post(url=url, data=payload)
-
-        if response.status_code == 200:
-            raw: str = response.text
-
-            # Unkludge the kludgefest "valid JSON" (lol) response from FireEye
-            objects = [komand.helper.clean(json.loads(l)) for l in raw.splitlines() if (not l.isnumeric() and len(l))]
-
-            return {Output.ALERTS: objects}
-        else:
-            try:
-                details = response.json()["details"][0]
-                code, message = details["code"], details["message"]
-
-                raise Exception(f"Error: {message} (error code {code}")
-
-            except JSONDecodeError as e:
-                raise Exception(
-                    f"Error: Malformed response received from FireEye HX appliance. " f"Got: {response.text}"
-                ) from e
-            except (KeyError, IndexError) as e:
-                raise Exception(
-                    "Error: Unknown error received from FireEye HX appliance " "(no error cause reported)."
-                ) from e
+        return {
+            Output.ALERTS: insightconnect_plugin_runtime.helper.clean(
+                self.connection.api.get_alerts_by_host_id(
+                    {"agent._id": params.get(Input.HOST_ID), "limit": 50, "offset": 0}
```

Comment:
Would be great to have those parameters as optional inputs 😃

Reply:
Because they are used in pagination to retrieve all alerts for a given host. So far, we have avoided adding such inputs, even as optional, to make the use of plugins easier and more user-friendly. Therefore, I think we shouldn't add `limit` and `offset` as optional inputs.

Reply:
@dsliwinski-r7  we dont support pagination in the product, if you limit it for 50, we will be returning only 50 all the time. We want the user to be able to search and fetch the alerts that they want in the case of over 50. Am I missing anything here?